examples/auth/schema.ts

```diff
@@ -1,67 +1,147 @@
 import { list } from '@keystone-6/core';
-import { allOperations, allowAll } from '@keystone-6/core/access';
+import { allowAll } from '@keystone-6/core/access';
 import { text, checkbox, password } from '@keystone-6/core/fields';
 import type { Lists } from '.keystone/types';
 
+type Session = {
+  itemId: string;
+  data?: {
+    isAdmin: boolean;
+  };
+}
+
+function hasSession({ session }: { session: Session | undefined }) {
+  return Boolean(session)
+}
+
+function isAdminOrSameUser({ session, item }: { session: Session | undefined; item: Lists.User.Item }) {
+  // you need to have a session to do this
+  if (!session) return false;
+
+  // admins can do anything
+  if (session.data?.isAdmin) return true;
+
+  // the authenticated user needs to be equal to the user we are updating
+  return session.itemId === item.id;
+}
+
+function isAdminOrSameUserFilter({ session }: { session: Session | undefined; }) {
+  // you need to have a session to do this
+  if (!session) return false;
+
+  // admins can see everything
+  if (session.data?.isAdmin) return {};
+
+  // the authenticated user can only see themselves
+  return {
+    id: {
+      equals: session.itemId
+    }
+  }
+}
+
+function isAdmin({ session }: { session: any }) {
+  // you need to have a session to do this
+  if (!session) return false;
+
+  // admins can do anything
+  if (session.data.isAdmin) return true;
+
+  // otherwise, no
+  return false;
+}
+
 export const lists: Lists = {
   User: list({
     access: {
       operation: {
-        ...allOperations(allowAll),
-        // Only allow admins to delete users
-        delete: ({ session }) => session?.data.isAdmin,
+        create: allowAll,
+        query: allowAll,
+
+        // only allow users to update _anything_, but what they can update is limited by
+        //   the access.filter.* and access.item.* access controls
+        update: hasSession,
+
+        // only allow admins to delete users
+        delete: isAdmin
+      },
+      filter: {
+        update: isAdminOrSameUserFilter
       },
+      item: {
```

Comment:
Is this doing the same thing as `filter` - maybe a better example of `item` access control is something like you can't update an email and a password at the same time??